runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -231,13 +247,17 @@ private static Object handleReadDataCorruption(CorfuPayloadMsg<ReadResponse> msg
      * completes.
      */
     public CompletableFuture<ReadResponse> read(long address) {
-        return router.sendMessageAndGetCompletable(
+        Timer.Context context = getTimerContext("read");
+        CompletableFuture<ReadResponse> cf = router.sendMessageAndGetCompletable(
                 CorfuMsgType.READ_REQUEST.payloadMsg(new ReadRequest(address)));
+        return cf.thenApply(x -> { context.stop(); return x; });
     }
 
     public CompletableFuture<ReadResponse> read(UUID stream, Range<Long> offsetRange) {
-        return router.sendMessageAndGetCompletable(
+        Timer.Context context = getTimerContext("readRange");
+        CompletableFuture<ReadResponse> cf = router.sendMessageAndGetCompletable(
                 CorfuMsgType.READ_REQUEST.payloadMsg(new ReadRequest(offsetRange, stream)));
+        return cf.thenApply(x -> { context.stop(); return x; });
```

Comment:
It wasn't clear when I started if that would be useful. Now, it still isn't crystal clear.  If we created a new flavor of `sendMessageAndGetCompletable` that had an extra argument (a closure to attach to the CompletableFuture using `.thenApply`), hm, we'd have to add that argument (optional or mandatory) to all functions "between" whereever we want to start a timer down to  `sendMessageAndGetCompletable`.